Added Makefile, MIS, CalculationBehaviour

EmilSoleymani opened the pull request:
# Makefile
## Testing
The `make test` rule now works with single `runtests.jl` file. Apparently make files execute each line from directory of Makefile, so combining the lines we previously had to `cp test && julia --project=@. runtests.jl` worked.
## Running
Also added a rule in the Makefile that runs the script. This `make run` is executed by default, so just executing `make` will run the vdisp module.
## Cleaning
Made several rules for cleaning up output files and LaTeX auxiliary files. `make clean` runs all cleaning rules, `make cleanDocs` only cleans any LaTeX aux files. `make cleanOutput` cleans any output files from VDisp.
# MIS
Added a preliminary draft of the MIS. Many sections are still marked as TODO, however three of the major modules, the ones that are already coded, have been specified. 
# CalculationBehaviour
Added all concrete implementations of CalculationBehaviour and its functions. The getValue() function of each implementation must be changed to actually perform corresponding calculations and return Tuples of values.

smiths reviewed (approved):
As we discussed before, please use a separate PR for each change.  You can have more than one PR open at once, they will just be on separate branches.

The `makefile` runs easily.  All tests passed.

Good to see you start the MIS.

EmilSoleymani commented:
> As we discussed before, please use a separate PR for each change. You can have more than one PR open at once, they will just be on separate branches.

@smiths  I will try to keep changes on separate branches and PRs moving forward. My main concern is with merge conflicts since I haven't previously worked on multiple branches simultaneously before. I have heard of merge conflicts but have never encountered one or seen how to resolve one. I have a few questions relating to this. In PR #60 I added exception handling to the `InputParser.jl` file. Now let's say I wanted to add a completely different functionality to that same file. My goal is to achieve this on a separate branch so I can make a separate PR. When I go to make a new branch, I should make this branch have the same history as `update-input-parser` branch instead of `main` branch to reflect my changes to `InputParser.jl` in PR #60 ? This way when I add more things to `InputParser.jl` it is not missing all the info that will be added. Or is it more appropriate to add the other changes to a version of the file that is the same as the one on `main` branch - which doesn't include any of the exception handling code yet - and deal with any merge conflicts when the time comes (Again since I have never seen merge conflicts I am only assuming there will be conflicts since I added many things to `InputParser.jl` in PR #60 that wouldn't be present in the other PR, while new lines not present in PR #60 would be present in the new PR).

smiths commented:
@EmilSoleymani, the answer to your question depends on the specifics.  If a branch, say B, is going to live a long time and in the other branch, say A, you are making a small change, you should branch A from main and make the change and get it merged back to main.  You would then merge main into your branch B.  It is good practice with long lived branches to frequently merge main into your branch, so that it stays reasonably up to date.  This isn't coming up much for you, since you are the principle developer for everything.  :smile:

If your new branch A depends on changes made in B, you could branch from B and merge A into B when it is done.  This would delay having the changes seen in main though.

It is also possible to [cherry pick commits](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/cherry-picking-a-commit).
